### ISSUE TITLE
[Bug]: Using official Image ID to create server causes it to be recreated every apply

### ISSUE THREAD

**Issue by apricote** [hetznercloud/terraform-provider-hcloud#657]
### What happened?

When using the ID of an official image (e.g. `ubuntu-22.04`) Terraform will always try to recreate the server, because we set the Image name instead of the image ID into the state.

Example with the config below:

```shell
➜ tf apply

data.hcloud_ssh_key.jt: Reading...
data.hcloud_image.image: Reading...
data.hcloud_ssh_key.jt: Read complete after 0s [name=julian.toelle]
data.hcloud_image.image: Read complete after 1s [name=ubuntu-22.04]
hcloud_server.node: Refreshing state... [id=31170104]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # hcloud_server.node must be replaced
-/+ resource "hcloud_server" "node" {
      + backup_window              = (known after apply)
      ~ datacenter                 = "fsn1-dc14" -> (known after apply)
      ~ firewall_ids               = [] -> (known after apply)
      ~ id                         = "31170104" -> (known after apply)
      ~ image                      = "ubuntu-22.04" -> "67794396" # forces replacement
      ~ ipv4_address               = "91.107.217.44" -> (known after apply)
      ~ ipv6_address               = "2a01:4f8:c010:aa09::1" -> (known after apply)
      ~ ipv6_network               = "2a01:4f8:c010:aa09::/64" -> (known after apply)
      - labels                     = {} -> null
        name                       = "recreate"
      ~ status                     = "running" -> (known after apply)
        # (8 unchanged attributes hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes 

hcloud_server.node: Destroying... [id=31170104]
hcloud_server.node: Destruction complete after 8s
hcloud_server.node: Creating...
hcloud_server.node: Creation complete after 9s [id=31170136]

Apply complete! Resources: 1 added, 0 changed, 1 destroyed.
```

### What did you expect to happen?

I expected the server to not be recreated every apply.

### Please provide a minimal working example

```hcl
data "hcloud_image" "image" {
  name = "ubuntu-22.04"
  with_architecture = "x86"
}

resource "hcloud_server" "node" {
  name        = "recreate"
  location    = "fsn1"
  server_type = "cx11"
  image       = data.hcloud_image.image.id
}
```

**Comment by mixman** [comment]
Thanks for looking into this! I keep receiving "Error: image 67794396 for architecture x86 not found" with a similar config.

Temporary workaround (to get any resources up; parent issue remains) using image ID instead of name:
`image       = "67794396" #ubuntu-22.04"`

**EDIT:** 1.38.1 fixes image = "ubuntu-22.04" usage. Remains broken for hcloud_image.

**Comment by korewaChino** [comment]
Also having this issue when using the named ID (`alma-9`)
Using the image ID (`105906519`) will cause the resource to always be tainted, because it already changed back to the named ID when applied

**Comment by apricote** [comment]
Fix was released as `v1.38.2` @korewaChino
